python/ray/data/dataset.py

```diff
@@ -622,7 +626,7 @@ def _equalize_larger_splits(
             leftovers = []
             for split in splits:
                 size = counts_cache[split._get_uuid()]
-                if size == target_size:
+                if size == target_size or target_size == 0:
```

Comment:
This isn't exactly a proper fix, but the simplest one for now. I am thinking of rewriting split() to clean up the logic.